core/phase.js

```diff
@@ -256,8 +247,15 @@ PhaseBase.prototype.impl1ToNAsync = function(stream) {
     t.end();
     return result.then(function(result) {
       flow.end();
+      return process();
     });
-  })).then(function() {
+  }
+  var spawn = Math.min(this.parallel, items.length);
+  var tasks = [];
+  for (var i = 0; i < spawn; i++) {
+    tasks.push(process());
+  }
+  return Promise.all(tasks).then(function() {
```

Comment:
can this spawning behaviour be factored out into a function? For the next CL, at any rate.